tests/unit/aiplatform/test_jobs.py

```diff
@@ -969,3 +977,93 @@ def test_batch_predict_job_with_versioned_model(
             ].model
             == _TEST_VERSIONED_MODEL_NAME
         )
+
+
+@pytest.fixture
+def create_mdm_job_mock():
+    with mock.patch.object(
+        _TEST_API_CLIENT, "create_model_deployment_monitoring_job"
+    ) as create_mdm_job_mock:
+        create_mdm_job_mock.return_value = (
+            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
+                name=_TEST_MDM_JOB_NAME,
+                display_name=_TEST_DISPLAY_NAME,
+                state=_TEST_JOB_STATE_RUNNING,
+            )
+        )
+        yield create_mdm_job_mock
+
+
+@pytest.fixture
+def get_mdm_job_mock():
+    with mock.patch.object(
+        _TEST_API_CLIENT, "get_model_deployment_monitoring_job"
+    ) as get_mdm_job_mock:
+        get_mdm_job_mock.return_value = (
+            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
+                name=_TEST_MDM_JOB_NAME,
+                display_name=_TEST_DISPLAY_NAME,
+                state=_TEST_JOB_STATE_RUNNING,
+                endpoint=_TEST_ENDPOINT,
+            )
+        )
+        yield get_mdm_job_mock
+
+
+@pytest.fixture
+def mock_deployed_model():
+    with mock.patch.object(gca_endpoint_compat, "DeployedModel") as mock_deployed_model:
+        mock_deployed_model.id = _TEST_ID
+        mock_deployed_model.explanation_spec.parameters = "TEST_VALUE"
+        yield mock_deployed_model
+
+
+@pytest.fixture
+def endpoint_list_models_mock(mock_deployed_model):
+    with mock.patch.object(
+        aiplatform.Endpoint, "list_models"
+    ) as endpoint_list_models_mock:
+        endpoint_list_models_mock.return_value = [mock_deployed_model]
+        yield endpoint_list_models_mock
```

Comment:
The underlying GAPIC clients should be mocked. The mocked client APIs should return an instance of the proto resource as expected instead of mocking the proto class. The mocks in the endpoint tests can be reused.

Reply:
update: I have removed the endpoint-related mocks and imported them from the endpoint unit tests. I have also opted to use adding drift detection configs as a way to test the update function instead of enabling/disabling XAI. Test coverage for XAI configs already exists in other model monitoring unit and system tests.